src/plugins/advanced_settings/public/header_user_theme_menu.tsx

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { ChangeEvent, SyntheticEvent, useState } from 'react';
+import { i18n } from '@osd/i18n';
+import { FormattedMessage } from '@osd/i18n/react';
+import {
+  EuiButton,
+  EuiCallOut,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFormRow,
+  EuiHeaderSectionItemButton,
+  EuiIcon,
+  EuiLink,
+  EuiPopover,
+  EuiPopoverTitle,
+  EuiSelect,
+  EuiSpacer,
+} from '@elastic/eui';
+import { CoreStart } from 'opensearch-dashboards/public';
+import { useOpenSearchDashboards, useUiSetting$ } from '../../opensearch_dashboards_react/public';
+
+export const HeaderUserThemeMenu = () => {
+  const {
+    services: {
+      http: { basePath },
+      uiSettings,
+    },
+  } = useOpenSearchDashboards<CoreStart>();
+  // TODO: move to central location?
+  const themeOptions = [
+    {
+      value: 'v7',
+      text: 'v7',
+    },
+    {
+      value: 'next',
+      text: 'Next (preview)',
+    },
+  ];
+  const screenModeOptions = [
+    {
+      value: 'light',
+      text: 'Light mode',
+    },
+    {
+      value: 'dark',
+      text: 'Dark mode',
+    },
+    {
+      value: 'automatic',
+      text: 'Use browser settings',
+    },
+  ];
+  const prefersAutomatic =
```

Comment:
So, the existing type signature for `theme:darkMode` is a boolean that indicates whether the current mode is light or dark. I decided that the safest and easiest approach was not to break/change that contract, because that's what all `.get` callers actually care about. Instead, preferring automatic is an alternative way to _set_ that setting, not a setting itself.